### ISSUE TITLE
NPE opening some caches

### ISSUE THREAD

**Issue by SammysHP** [cgeo/cgeo#15526]
### Describe your problem!

Just tried the latest nightly on an Android 7 device and got following problem:

Try to open cache details of some (but not all) caches leads to a NPE.

### How to reproduce?

Open https://coord.info/GCAAD9X

### Actual result after these steps?

```
java.lang.NullPointerException: Attempt to read from field 'int android.graphics.Bitmap$Config.nativeInt' on a null object reference
	at android.graphics.Bitmap.createBitmap(Bitmap.java:879)
	at android.graphics.Bitmap.createBitmap(Bitmap.java:856)
	at android.graphics.Bitmap.createBitmap(Bitmap.java:823)
	at androidx.core.graphics.BitmapCompat.createScaledBitmap(BitmapCompat.java:295)
	at cgeo.geocaching.utils.ImageUtils.scaleBitmapTo(ImageUtils.java:215)
	at cgeo.geocaching.utils.ImageUtils.scaleBitmapToDisplay(ImageUtils.java:194)
	at cgeo.geocaching.network.HtmlImage.scaleImage(HtmlImage.java:350)
	at cgeo.geocaching.network.HtmlImage$1.loadFromDisk(HtmlImage.java:301)
	at cgeo.geocaching.network.HtmlImage$1.lambda$subscribe$1(HtmlImage.java:283)
	at cgeo.geocaching.network.HtmlImage$1.$r8$lambda$UzFeLQ5bjonuPs5zXj6tkdY7-NY(HtmlImage.java:0)
	at cgeo.geocaching.network.HtmlImage$1$$ExternalSyntheticLambda1.run(R8$$SyntheticClass:0)
	at io.reactivex.rxjava3.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:38)
	at io.reactivex.rxjava3.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:25)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:272)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
	at java.lang.Thread.run(Thread.java:761)
```

### Reproducible

Yes

### c:geo Version

2024.03.27-NB-ac81856

### System information

```text
## System information

c:geo version: 2024.03.27-NB-ac81856

Device:
-------
- Device type: Redmi Note 4 (lineage_mido, Xiaomi)
- Available processors: 8
- Android version: 7.1.2
- Android build: lineage_mido-userdebug 7.1.2 NJH47F 57ca1c7c3e
- Screen resolution: 1080x1920px (400x687dp)
- Pixel density: 2.7
- System font scale: 1.0 / used scale: 1.0
- Sailfish OS detected: false
- Google Play services: disabled - 24.08.12 (040400-608507424)
- Memory:  Available:929,7 MB, Total:2,8 GB, Threshold: 216,0 MB, low:false

Sensor and location:
-------
- Low power mode: inactive
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: rotation vector

Program settings:
-------
- Settings: v9, Count:235
- Set language: de_DE (system default)
- System date format: dd.MM.yy
- Time zone: GMT+01:00
- Debug mode active: no
- Log Settings: debug=false, minLevel=WARN, minAddCallerInfo=NONE, addCallerInfoMaxDepth=8, throwOnError=false, transactionSizes=false
- Last manual backup: never
- Last auto backup: 6. März 2022, 15:51
- Routing mode: Straight
- Live map mode: true
- Use unified map: true
- OSM multi-threading: true / threads: 4
- Map: OpenStreetMap.org
  - Id: cgeo.geocaching.maps.mapsforge.MapsforgeMapProvider$OsmMapSource
  - Atts: © OpenStreetMap DE, map data OpenStreetMap contributors
  - Theme: fzk-outdoor-contrast-v5-hillshading.zip:fzk-outdoor-contrast-v5-hillshading/fzk-outdoor-contrast-v5.xml

Filters:
-------
- Hide waypoints: -
- LIVE: Schwierigkeit / Gelände: ≤4.0 / ≤4.0, Status: 3 Elemente ({"advanced":true,"inconclusive":true,"tree":{"type":"AND","children":[{"type":"type","config":{}},{"type":"difficulty_terrain","config":{"d":{"max":4.0},"t":{"max":4.0}}},{"type":"status","config":{"values":["owned_no","found_no","exclude_archived"]}}]}})
- OFFLINE: Keine ({"advanced":false,"inconclusive":false,"tree":{"type":"AND","children":[{"type":"type","config":{}},{"type":"difficulty_terrain","config":{"d":{},"t":{}}},{"type":"status","config":{}}]}})
- Additional stored filters: 1

Services:
-------
- Geocaching sites enabled:
   - geocaching.com: Logged in (Anmeldung OK) / PREMIUM
   - opencaching.de: Logged in (Anmeldung OK)
- Geocaching.com date format: MM/dd/yyyy
- Geocaching.com website language: en-US
- Last login error on geocaching.com: communication error (2024-03-27 13-43)
- Last successful login on geocaching.com: 2024-03-27 16-28
- Routing: external / BRouter installed: true

Permissions
-------
- ACCESS_COARSE_LOCATION: granted
- ACCESS_FINE_LOCATION: granted
- READ_CONTACTS: DENIED
- WRITE_EXTERNAL_STORAGE: granted
- POST_NOTIFICATIONS: granted

Paths
-------
- System internal c:geo dir: /data/user/0/cgeo.geocaching (2,6 GB free) v3 internal isDir(13 entries)
- Legacy User storage c:geo dir: /storage/emulated/0/cgeo (2,6 GB free) v3 external non-removable isDir(9 entries)
- Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (2,6 GB free) v3 external non-removable isDir(35 entries)
- Internal theme sync (is turned ON): /data/user/0/cgeo.geocaching/MapThemeData (2,6 GB free) v3 internal isDir(6 entries)
- Map render theme path: fzk-outdoor-contrast-v5-hillshading.zip:fzk-outdoor-contrast-v5-hillshading/fzk-outdoor-contrast-v5.xml
- Public Folders: #13
  - BASE: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo (av:true, files:>=70, size:>=27,7 MB, free:>=2,6 GB)
  - OFFLINE_MAPS: content://com.android.externalstorage.documents/tree/primary:Locus/mapsVector/document/primary:Locus/mapsVector (av:true, files:>=102, size:>=324,8 MB, free:>=2,6 GB)
  - OFFLINE_MAP_THEMES: content://com.android.externalstorage.documents/tree/primary:cgeo/themes/document/primary:cgeo/themes (av:true, files:>=5, size:>=1,7 MB, free:>=2,6 GB)
  - OFFLINE_MAP_SHADING: content://com.android.externalstorage.documents/tree/primary:Locus/data/srtm/document/primary:Locus/data/srtm (av:true, files:>=17, size:>=46,0 MB, free:>=2,6 GB)
  - LOGFILES: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/logfiles (av:true, files:>=0, size:>=0 B, free:>=2,6 GB)
  - GPX: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/gpx (av:true, files:>=64, size:>=26,0 MB, free:>=2,6 GB)
  - BACKUP: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/backup (av:true, files:>=4, size:>=1,1 MB, free:>=2,6 GB)
  - FIELD_NOTES: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/field-notes (av:true, files:>=1, size:>=206 B, free:>=2,6 GB)
  - SPOILER_IMAGES: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/GeocachePhotos (av:true, files:>=0, size:>=0 B, free:>=2,6 GB)
  - ROUTING_BASE: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/routing (av:true, files:>=8, size:>=81,6 KB, free:>=2,6 GB)
  - ROUTING_TILES: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/routing/segments4 (av:true, files:>=0, size:>=0 B, free:>=2,6 GB)
  - WHERIGO: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/wherigo (av:true, files:>=0, size:>=0 B, free:>=2,6 GB)
  - TEST_FOLDER: file:///data/user/0/cgeo.geocaching/files/unittest (av:true, files:>=0, size:>=0 B, free:>=2,6 GB)
- PersistedDocumentUris: #2
  - PROXIMITY_NOTIFICATION_FAR: null
  - PROXIMITY_NOTIFICATION_CLOSE: null
- Persisted Uri Permissions: #4
  - content://com.android.externalstorage.documents/tree/primary:Locus/mapsVector (20. Feb. 2021, 09:09):RW
  - content://com.android.externalstorage.documents/tree/primary:Locus/data/srtm (5. Nov. 2022, 11:20):RW
  - content://com.android.externalstorage.documents/tree/primary:cgeo (20. Feb. 2021, 09:08):RW
  - content://com.android.externalstorage.documents/tree/primary:cgeo/themes (9. Mai 2021, 07:50):RW
- Database: /data/user/0/cgeo.geocaching/databases/data (v103, Size:1,0 MB) on system internal storage

--- End of system information ---
```

**Comment by eddiemuc** [comment]
the only image I found in the example cache listing was of www.gc-apps.com. I set up an emulator for SDK24. c:geo on this emulator could NOT download the image because of a SSL validation problem for that server:

`[OkHttp] HTTP-ERROR: (63 ms) GET https://www.gc-apps.com/checker/66bcad458a4659292d23728f3135bc24/image (javax.net.ssl.SSLHandshakeException: java.security.cert.CertPathValidatorException: Trust anchor for certification path not found.) javax.net.ssl.SSLHandshakeException: java.security.cert.CertPathValidatorException: Trust anchor for certification path not found.`
              
I assume that root certificates are not up-to-date on the (fresh) emulator image...

However, c:geo did not throw an NPE on this but correcly dispayed the image as "not downloadable" in the gallery.
                                                 


I suspect the root cause to be a certification validation problem with the certificate of www.gc-apps.com (the only image in the example cache listing was of this server,

**Comment by eddiemuc** [comment]
Root cause is that bitmap in question don't have a config `(getConfig()` returns null), and legacy SDKs need this to be non-null. BitmapCompat seems to behave incomplete here, because newer SDKs allow `getConfig()` to return null. 

Question remains why for the bitmap in the NPE the config should be null.

Odd idea: is it possible this NPE happens whenever there's a SVG image on the page? Those are handled special by c:geo.

**Comment by Lineflyer** [comment]
Same problem for me on API21 emulator:
- I stored GC1H0H0 from map (sheet)
- When trying to open it from the list of stored caches cgeo crashes with NPE

```
2024-04-03 17:18:03.075  4864-4901  Andro...time cge...developer  E  FATAL EXCEPTION: RxComputationThreadPool-3
                                                                     Process: cgeo.geocaching.developer, PID: 4864
                                                                     java.lang.NullPointerException: Attempt to read from field 'int android.graphics.Bitmap$Config.nativeInt' on a null object reference
                                                                     	at android.graphics.Bitmap.createBitmap(Bitmap.java:817)
                                                                     	at android.graphics.Bitmap.createBitmap(Bitmap.java:794)
                                                                     	at android.graphics.Bitmap.createBitmap(Bitmap.java:761)
                                                                     	at androidx.core.graphics.BitmapCompat.createScaledBitmap(BitmapCompat.java:295)
                                                                     	at cgeo.geocaching.utils.ImageUtils.scaleBitmapTo(ImageUtils.java:215)
                                                                     	at cgeo.geocaching.utils.ImageUtils.scaleBitmapToDisplay(ImageUtils.java:194)
                                                                     	at cgeo.geocaching.network.HtmlImage.scaleImage(HtmlImage.java:350)
                                                                     	at cgeo.geocaching.network.HtmlImage$1.loadFromDisk(HtmlImage.java:301)
                                                                     	at cgeo.geocaching.network.HtmlImage$1.lambda$subscribe$1$cgeo-geocaching-network-HtmlImage$1(HtmlImage.java:283)
                                                                     	at cgeo.geocaching.network.HtmlImage$1$$ExternalSyntheticLambda1.run(D8$$SyntheticClass)
                                                                     	at io.reactivex.rxjava3.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:38)
                                                                     	at io.reactivex.rxjava3.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:25)
                                                                     	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                                                                     	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:152)
                                                                     	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:265)
                                                                     	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
                                                                     	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
                                                                     	at java.lang.Thread.run(Thread.java:818)
```

**Comment by Lineflyer** [comment]
The listing of GC1H0H0 is quite simple. The only suspicious element would be a GIF image?!

**Comment by eddiemuc** [comment]
Hopefully fixed with #15548
